util/eventbus/subscribe.go

```diff
@@ -81,16 +94,74 @@ func (q *Queue) pump(stop goroutineShutdownWorker) {
 	}
 }
 
-// A Subscriber delivers one type of event from a [Queue].
+func (s *subscribeState) addSubscriber(t reflect.Type, sub subscriber) {
+	s.outputsMu.Lock()
+	defer s.outputsMu.Unlock()
+	if s.outputs[t] != nil {
+		panic(fmt.Errorf("double subscription for event %s", t))
+	}
+	s.outputs[t] = sub
+	s.client.addSubscriber(t, s)
+}
+
+func (s *subscribeState) deleteSubscriber(t reflect.Type) {
+	s.outputsMu.Lock()
+	defer s.outputsMu.Unlock()
+	delete(s.outputs, t)
+	s.client.deleteSubscriber(t, s)
+}
+
+func (q *subscribeState) subscriber(val any) subscriber {
+	q.outputsMu.Lock()
+	defer q.outputsMu.Unlock()
+	return q.outputs[reflect.TypeOf(val)]
+}
+
+// Close closes the subscribeState. Implicitly closes all Subscribers
```

Comment:
Need this be exported? (Harmless, but I think not used outside the client, which  has its own Close for the user to touch)

Reply:
Hmm, no need to export yeah, fixed. I think it was just muscle memory, I don't see private close that often :)